src/main/java/utility/TaskList.java

```diff
@@ -0,0 +1,153 @@
+            else {
+                return this.tasks.get(this.tasks.size() - 1);
+            }
+        }
+        else {
+            return null;
+        }
+    }
+    
+    /**
+     * Returns a string representing all tasks in the list, each numbered sequentially.
+     *
+     * @return a formatted string of all tasks in the list
+     */
+    public String listWord() {
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < this.tasks.size(); i++) {
+            result.append(i + 1).append(". ")
+                .append(this.tasks.get(i).toString()).append("\n");
+        }
+        return result.toString();
+    }
+    
+    /**
+     * Modifies the completion status of the task at the specified index.
+     *
+     * @param index the position of the task in the list (1-based index)
+     * @param markStatus {@code true} to mark the task as done, {@code false} to mark it as undone
+     * @return a string representing the modified task
+     */
+    public String modifyOperation(int index, boolean markStatus) {
+        int normalizedIndex = index - 1;
+        StringBuilder result = new StringBuilder();
+        if (markStatus) {
+            this.tasks.get(normalizedIndex).markAsDone();
+        } else {
+            this.tasks.get(normalizedIndex).markAsUndone();
+        }
+        result.append(this.tasks.get(normalizedIndex).toString());
+        return result.toString();
+    }
+    
+    /**
+     * Deletes the task at the specified index from the list.
+     *
+     * @param index the position of the task in the list (1-based index)
+     * @return a string representing the deleted task
+     */
+    public String deleteOperation(int index) {
+        int normalizedIndex = index - 1;
+        StringBuilder result = new StringBuilder();
+        result.append(this.tasks.get(normalizedIndex).toString());
+        this.tasks.remove(normalizedIndex);
+        return result.toString();
+    }
+    
+    /**
+     * Returns a string indicating the number of tasks currently in the list.
+     *
+     * @return a formatted string showing the current number of tasks
+     */
+    public String getLength() {
+        return "Now you have " + String.valueOf(this.tasks.size()) + " tasks in the list.";
+    }
+    
+    /**
+     * Returns the list of tasks managed by this {@code TaskList}.
+     *
+     * @return an {@code ArrayList} of tasks
+     */
+    public ArrayList<Task> getTaskLists() {
+        return this.tasks;
+    };
+    
+    /**
+     * Finds and returns a list of tasks that match the specified search parameter.
+     * The search is case-insensitive and matches the search parameter against the
+     * description of each task in the list.
+     *
+     * @param searchParameter the keyword or phrase to search for within the task descriptions
+     * @return an {@code ArrayList} of tasks that match the search criteria
+     */
+    public ArrayList<Task> findLists(String searchParameter) {
```

Comment:
```suggestion
    public ArrayList<Task> findItemsInList(String searchParameter) {
```
`findLists` might be misleading as it describes there exists multiple lists. Perhaps `findItemsInList` might be more suitable?